R/zchunk_L2072.ag_water_irr_mgmt.R

```diff
@@ -1,67 +1,322 @@
+    A03.sector <- get_data(all_data, "water/A03.sector")
+
+    # Primary crops
+    # Compute irrigation water consumption IO coefficients (km3/Mt crop = m3/kg) by region / irrigated crop / year / GLU / management level
+    L165.BlueIrr_m3kg_R_C_GLU %>%
+      # Initial adjustment: drop any water coefs for region/crop/GLU combinations that aren't written out
+      semi_join(L161.ag_irrProd_Mt_R_C_Y_GLU, by = c("GCAM_region_ID", "GCAM_commodity", "GLU")) %>%
+      mutate(IRR_RFD = "IRR", water_type = "water consumption") %>%
+      rename(value = BlueIrr_m3kg) ->
+      L2072.Blue_IRR_IO_R_C_GLU
+
+    # Compute biophysical water consumption IO coefficients (km3/Mt crop = m3/kg) by region / irrigated crop / year / GLU / management level
+    L165.TotIrr_m3kg_R_C_GLU %>%
+      # Initial adjustment: drop any water coefs for region/crop/GLU combinations that aren't written out
+      semi_join(L161.ag_irrProd_Mt_R_C_Y_GLU, by = c("GCAM_region_ID", "GCAM_commodity", "GLU")) %>%
+      mutate(IRR_RFD = "IRR", water_type = "biophysical water consumption") %>%
+      rename(value = TotIrr_m3kg) ->
+      L2072.Bio_IRR_IO_R_C_GLU
+
+    # Compute biophysical water consumption IO coefficients (km3/Mt crop = m3/kg) by region / rainfed crop / year / GLU / management level
+    L165.GreenRfd_m3kg_R_C_GLU %>%
+      # Initial adjustment: drop any water coefs for region/crop/GLU combinations that aren't written out
+      semi_join(L161.ag_rfdProd_Mt_R_C_Y_GLU, by = c("GCAM_region_ID", "GCAM_commodity", "GLU")) %>%
+      mutate(IRR_RFD = "RFD", water_type = "biophysical water consumption") %>%
+      rename(value = GreenRfd_m3kg) ->
+      L2072.Bio_RFD_IO_R_C_GLU
+
+    # Following are repeated processing steps for the three files, so combine them all togather
+    L2072.Blue_IRR_IO_R_C_GLU %>%
+      bind_rows(L2072.Bio_IRR_IO_R_C_GLU, L2072.Bio_RFD_IO_R_C_GLU) %>%
+      left_join_error_no_match(GCAM_region_names, by = "GCAM_region_ID") %>%
+      left_join_error_no_match(basin_to_country_mapping[c("GLU_code", "GLU_name")], by = c("GLU" = "GLU_code")) %>%
+      # Copy to both high and low management levels
+      repeat_add_columns(tibble::tibble(MGMT = c("hi", "lo"))) %>%
+      # Add sector, subsector, technology names
+      mutate(AgSupplySector = GCAM_commodity,
+             AgSupplySubsector = paste(GCAM_commodity, GLU_name, sep = "_"),
+             AgProductionTechnology = paste(GCAM_commodity, GLU_name, IRR_RFD, MGMT, sep = "_"),
+             coefficient = round(value, aglu.DIGITS_CALOUTPUT)) %>%
+      filter(coefficient > 0) %>%
+      # Assume coefs stay constant, copy to all model years
+      repeat_add_columns(tibble::tibble(year = MODEL_YEARS)) ->
+      L2072.AgCoef_Water_ag_mgmt
+
+    # Separate irrigation water consumption IO coefficients
+    L2072.AgCoef_Water_ag_mgmt %>%
+      filter(water_type == "water consumption") %>%
+      # Standardize irrigation water input names
+      mutate(water_sector = "Irrigation",
```

Comment:
Hi @rynacui - can you clarify? Why something take 10x longer? I just want to understand what you've tried, and what you haven't. Thanks.